Implement Car and Driver Repositories, Service Layer, and Add more DTOs

aremm23 opened the pull request:
Created repositories for Car and Driver entities to manage database interactions. Implemented the service layer to handle business logic related to car and driver operations, including assigning and removing cars from drivers. Added more Data Transfer Objects (DTOs).

aremm23 committed:
feature/driver: add necessary dtos and mapper config

feature/driver: impl driver and car services

feature/driver: init service layer

feature/driver: add repository level for car and driver entities

aleshkey commented on driver-service/src/main/java/com/artsem/api/driverservice/service/CarService.java:
```diff
@@ -0,0 +1,24 @@
+
+    CarResponseDto getOne(Long id);
+
+    List<CarResponseDto> getMany(List<Long> ids);
```
для возвращения списка лучше сделать отдельную dto, где вот этот лист будет полем
ещё, если я не ошибаюсь, там нужна паггинация

aremm23 committed:
feature/driver: implement getList methods to get Page of filtered entities

feature/driver: add getList methods to get Page of filtered entities

feature/driver: add car and driver filters

fix/driver: fix car repository

aremm23 committed:
feature/driver: add ListResponseDto to return list of entities

aleshkey commented on driver-service/src/main/java/com/artsem/api/driverservice/config/PassengerMapperConfig.java:
```diff
@@ -0,0 +1,13 @@
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+public class PassengerMapperConfig {
```
Точно Passenger?

aleshkey commented on driver-service/src/main/java/com/artsem/api/driverservice/filter/DriverFilter.java:
```diff
@@ -0,0 +1,29 @@
+
+public record DriverFilter(String surnameLike, Boolean isFree) {
+    public Specification<Driver> toSpecification() {
+        return Specification.where(surnameLikeSpec()).and(isFreeSpec());
```
я бы оформил так, как ты оформлял стримы (.and с новой строки)

aleshkey commented on driver-service/src/main/java/com/artsem/api/driverservice/model/dto/responce/ListResponseDto.java:
```diff
@@ -0,0 +1,13 @@
+
+@AllArgsConstructor
+@Data
+public class ListResponseDto<T> {
```
а чего не рекорд? вроде не мапаешь. Ещё можно навесить ломбоковскую Builder для большей читаемости

aremm23 committed:
refactor/driver: cleanup code

refactor/driver: add @Builder annotations to dto classes

aleshkey commented on driver-service/src/main/java/com/artsem/api/driverservice/service/impl/CarServiceImpl.java:
```diff
@@ -0,0 +1,124 @@
+        List<CarResponseDto> dtosList = cars.stream()
+                .map(car -> mapper.map(car, CarResponseDto.class))
+                .toList();
+        return new ListResponseDto<>(dtosList.size(), dtosList);
```
так раз навесил билдер, то тут и использовать можно)

aremm23 committed:
refactor/driver: cleanup code
